components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -158,7 +141,13 @@ func (k *KnativeLib) GetChannelByLabels(namespace string, labels map[string]stri
 		log.Printf("ERROR: GetChannelByLabels(): channel list has %d items", channelListLength)
 		return nil, errors.New("length of channel list is not equal to 1")
 	}
-	return &channelList.Items[0], nil
+
+	channel := &channelList.Items[0]
+
+	if !channel.Status.IsReady() {
+		return nil, fmt.Errorf("ERROR: GetChannel():channel NotReady")
```

Comment:
Why not returning the channel even if it is not ready? The signature contract states that.

Reply:
This is because if an unready channel is returned then publishing to this channel will cause the dispatcher to return HTTP 500.

Reply:
I'd say checking for the status/readiness is then the responsibility of the client/caller.

Reply:
Right, I moved the check to the caller.